### PR TITLE
Remove obsolete bss-related code/build features

### PR DESCRIPTION
**Pull request by stinos**
GC for unix/windows builds doesn't make use of the bss section anymore,
so we do not need the (sometimes complicated) build features and code related to it.

Note for the esp8266 gc of bss is still used, not sure if that is needed?

### CODE REVIEW

File: py/gc.c [hunk 1]
```diff
@@ -693,34 +693,6 @@ void gc_dump_alloc_table(void) {
         int c = ' ';
         switch (ATB_GET_KIND(bl)) {
             case AT_FREE: c = '.'; break;
-            /* this prints out if the object is reachable from BSS or STACK (for unix only)
```
Comment (dpgeorge) [inline py/gc.c]: I would like to keep this for debugging purposes.  But, yes, would need to change the bss stuff to just scan the root pointer section of the mp_state_ctx structure.

**Comment by dpgeorge** [comment]
Still need to do this for esp8266 and teensy ports, but that should be a separate commit.

File: py/gc.c (hunk 1 above)
Comment (stinos) [inline py/gc.c]: If you keep this, it seems better to discard this PR and keep everything as-is well as it keeps debugging available not just for unix but also for mac/windows builds.

Comment (dpgeorge) [inline py/gc.c]: No, I like your code simplification :)  This code here will change to scan the root pointers using the global state structure, not using the bss symbols.  Therefore it'll become available on mac/windows as well, even with your code removal.

File: unix/Makefile [hunk 2]
```diff
@@ -36,7 +36,7 @@ endif
 ifeq ($(UNAME_S),Darwin)
 CC = clang
 # Use clang syntax for map file and set osx specific flags
```
Comment (blmorris) [inline unix/Makefile]: Maybe remove the "and set osx specific flags" part of this comment since it is no longer relevant?

Comment (stinos) [inline unix/Makefile]: Yes good point, thanks

File: py/gc.c (hunk 1 above)
Comment (stinos) [inline py/gc.c]: Ok I gave it a try, see updated comitted

**Comment by dpgeorge** [comment]
Thanks!

**Pull request merged**